src/components/project_view/submission_detail/submission_detail_panel.vue

```diff
@@ -0,0 +1,290 @@
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+
+import CorrectnessIcon, { CorrectnessLevel } from "@/components/project_view/submission_detail/correctness_icon.vue";
+
+export enum PanelCorrectnessLevel {
+    not_available = "not-available",
+    none_correct = 'none-correct',
+    some_correct = 'some-correct',
+    all_correct = 'all-correct',
+    info_only = 'info-only',
+    output_only = 'output-only',
+    dont_check = 'dont-check'
+}
+
+@Component({
+  components: {
+    CorrectnessIcon
+  }
+})
+export default class SubmissionDetailPanel extends Vue {
+  @Prop({required: true, type: String})
+  name!: string;
+
+  @Prop({required: true, type: String})
+  correctness_level!: CorrectnessLevel;
+
+  @Prop({default: 0, type: Number})
+  points_awarded!: number;
+
+  @Prop({default: 0, type: Number})
+  points_possible!: number;
+
+  @Prop({default: false, type: Boolean})
+  panel_is_active!: boolean;
+
+  @Prop({default: false, type: Boolean})
+  is_command!: boolean;
+
+  @Prop({default: false, type: Boolean})
+  is_multi_command_case!: boolean;
+
+  d_correctness_level: CorrectnessLevel = CorrectnessLevel.none_correct;
+  d_panel_is_active = false;
+  d_is_open = false;
+
+  @Watch('panel_is_active')
+  on_panel_is_active_change(new_value: boolean, old_value: boolean) {
+    this.d_panel_is_active = new_value;
+    this.d_is_open = this.d_panel_is_active;
+  }
+
+  @Watch('correctness_level')
+  on_correctness_level_change(new_value: CorrectnessLevel, old_value: CorrectnessLevel) {
+    this.d_correctness_level = new_value;
+  }
+
+  created() {
+    this.d_correctness_level = this.correctness_level;
+    this.d_panel_is_active = this.panel_is_active;
+    this.d_is_open = this.d_panel_is_active;
+  }
+
+  get panel_correctness() {
+    if (this.correctness_level === CorrectnessLevel.not_available) {
+      return PanelCorrectnessLevel.not_available;
+    }
+    else if (this.correctness_level === CorrectnessLevel.all_correct) {
+      return PanelCorrectnessLevel.all_correct;
+    }
+    else if (this.correctness_level === CorrectnessLevel.none_correct) {
+      return PanelCorrectnessLevel.none_correct;
+    }
+    else if (this.correctness_level === CorrectnessLevel.info_only) {
+      return PanelCorrectnessLevel.info_only;
+    }
+    return PanelCorrectnessLevel.some_correct;
+  }
+
+  toggle_d_is_open() {
+    if (this.correctness_level !== CorrectnessLevel.not_available) {
+      this.d_is_open = !this.d_is_open;
+    }
+  }
+}
+</script>
+
+<style scoped lang="scss">
+@import '@/styles/colors.scss';
+
+@mixin panel_header($background-color: $white-gray, $border-color: $white-gray,
+                    $cursor-option: pointer) {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-content: center;
+  background-color: $background-color;
+  border: 1px solid $border-color;
+  font-weight: bold;
+  cursor: $cursor-option;
+}
```

Comment:
Here's a fun scss trick you can do (you don't have to make this change, just letting you know it's a possibility):

```
@mixin panel_header($background-color: $white-gray, $border-color: $white-gray,
                    $cursor-option: pointer) {
  ...

  // The & gives you the parent selector
  &:hover {
    background-color: darken($background-color, 5%);
    border-color: darken($border-color, 5%);
  }
}
```

Then you don't need the second mixin or the explicit :hover classes later on